draft-ietf-tls-dtls-connection-id.md

```diff
@@ -303,14 +305,45 @@ type
 version
 : This value contains the version number. 
 
+cid
+: Value of the negotiated CID.
+
+cid_length
+: 1 byte field indicating the length of the negotiated CID.
+
 length
 : This value contains the length information in the outer-header. 
 
+## CBC Block Ciphers
+
+This  specification updates the MAC calculation defined in 
+Section 6.2.3.2 of {{RFC5246}}, which is re-used by Section
+4.1.2.3 of {{RFC6347}}. 
+
+The MAC is generated as:
+
+   MAC(MAC_write_key, seq_num +
+                        TLSCompressed.type +
+                        TLSCompressed.version +
+						cid +
+						cid_length + 
+                        TLSCompressed.length +
+                        TLSCompressed.fragment);
```

Comment:
Hi Hannes

given the TLS presentation language typically doesn't add clarity, can I suggest a different approach?  Could we just say that 'cid' and 'cid_length' are inserted after the 'DTLSCiphertext.length' and before whatever comes after it in the MAC computations defined by RFC 6347 and RFC 7366 (EtM), and similarly to the authentication data defined in  RFC 6347?

Reply:
When you agree on an approach please post to WGLC thread on the list.

Reply:
I believe I actually did it wrong. I should have only extended the MAC calculation in RFC 7366.

Reply:
Since we only define this for DTLS 1.2 the calculation should be: 

   MAC(MAC_write_key, seq_num +
       TLSCipherText.type +
       TLSCipherText.version +
       cid + 
       cid_len + 
       TLSCipherText.length +
       IV +
       ENC(content + padding + padding_length));

Correct?

Reply:
My concern is still:
This draft uses terms/definitions (`DTLSPlaintext`, `DTLSInnerPlaintext`, and `DTLSCiphertext`) and I would prefer to stay with these terms instead of cite new terms (and details) of other RFC's.

> Could we just say that 'cid' and 'cid_length' are inserted between DTLSCiphertext.version and DTLSCiphertext.length in the MAC computations 

That would be a generic solution, but I'm a little skeptical, which case this would cover, what the approach of this PR, adding the cid to the previous definitions, doesn't.

Reply:
@ekr (sorry, I can not resist:)
 
and you're still sure, that this discussions about including the cid into the MAC for 

https://github.com/tlswg/dtls-conn-id/issues/25#issuecomment-444890710

> Yes, but it might be true with some extension in the future. Moreover, authenticating the whole header is just good practice. 

pays off?

Reply:
I'm not sure why you think this is a useful question.

Reply:
> I'm not sure why you think this is a useful question.

In #25 I tried to point out, that in my opinion, changing the MAC is not required. The final answer was, it's good practice. So the effort of all the discussions about that is paid for good practice. In my opinion, when good practice gets that expensive, the question, if the good practice really pays off, is a valid question. I was interested in your opinion about that, therefore I asked.

Reply:
Opening with "I can not resist" hardly gives the impression that this is a legitimate question rather than trolling. In any case, I don't think this is particularly expensive and my opinion is unchanged.